### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by Saransh-cpp**
1. Add missing tests for all the files.
2. Update the files with tweeting capabilities so that they don't tweet when the tests are running.
3. Vary `C_rate` and not `"Current function [A]"` and `"Upper voltage cut-off [V]"`.
4. Update the workflow to read API keys when running tests on a PR.
5. A file restructure to make imports work and to reduce files in the root directory.

TODO:

- [x]  Add test for `random_plot_generator`.
- [x]  Add test for `information`.
- [x]  Making lint test (`flake8`) a requirement.
- [x]  Add `docstrings`

Note: The coverage will go down a bit because the tests cannot run on the lines which are responsible for tweeting, we can't tweet every time someone opens a PR or someone pushes a new commit to their existing PR.

### CODE REVIEW

**Comment by Saransh-cpp** [comment]
@tinosulzer, @rtimms, @brosaplanella could you please go through and review the `add-tests` branch whenever you get some time, I was thinking of keeping this as the basic structure which might change with a few minor tweaks as we add more code. I will start working on the degradation options and summary variables in a different PR (as discussed in the meet).

File: bot/information/information.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        solver: pybamm.BaseSolver
+        isExperiment: bool
+        cycle: list
+        number: numerical
+        isComparison: bool
+    Returns
+        tweet_text: str
+    """
+
+    if isExperiment:
```
Comment (tinosulzer) [inline bot/information/information.py:24]: use `lowercase_with_underscore` for variable names

File: bot/information/information.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            + " * "
+            + str(number)
+            + " "
+            + str(model)
+            + " "
+            + str(chemistry["citation"])
+        )
+        return tweet_text
+
+    elif isComparison:
```
Comment (tinosulzer) [inline bot/information/information.py:36]: use lowercase

File: bot/models/model_generator.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        sim: pybamm.Simulation
+        solution: pybamm.Simulation.solution
+    """
+
+    parameter_values = pybamm.ParameterValues(chemistry=chemistry)
+
+    parameter_values["Lower voltage cut-off [V]"] = lower_voltage
+    parameter_values["Ambient temperature [K]"] = ambient_temp
+    parameter_values["Initial temperature [K]"] = initial_temp
+    parameter_values["Reference temperature [K]"] = reference_temp
```
Comment (tinosulzer) [inline bot/models/model_generator.py:38]: reference temp defines at what temperature the parameters were measured and should not be changed

File: bot/plotting/random_plot_generator.py [hunk 4]
```diff
@@ -23,20 +45,17 @@ def random_plot_generator():
 
             modelNum = random.randint(0, len(models) - 1)
```
Comment (tinosulzer) [inline bot/plotting/random_plot_generator.py:46]: lowercase

File: bot/plotting/random_plot_generator.py [hunk 5]
```diff
@@ -23,20 +45,17 @@ def random_plot_generator():
...
             chemistries = [
+                pybamm.parameter_sets.Ai2020,
                 pybamm.parameter_sets.Chen2020,
                 pybamm.parameter_sets.Marquis2019,
                 pybamm.parameter_sets.Ecker2015,
-                pybamm.parameter_sets.Ramadass2004
+                pybamm.parameter_sets.Ramadass2004,
             ]
 
             chemNum = random.randint(0, len(chemistries) - 1)
```
Comment (tinosulzer) [inline bot/plotting/random_plot_generator.py:57]: lowercase

File: bot/plotting/random_plot_generator.py [hunk 6]
```diff
@@ -45,56 +64,55 @@ def random_plot_generator():
 
             solverNum = random.randint(0, len(solvers) - 1)
```
Comment (tinosulzer) [inline bot/plotting/random_plot_generator.py:65]: lowercase

File: bot/information/information.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        tweet_text: str
+    """
+
+    if isExperiment:
+        tweet_text = (
+            str(cycle)
+            + " * "
+            + str(number)
+            + " "
+            + str(model)
```
Comment (tinosulzer) [inline bot/information/information.py:30]: use `model.name` or open a PR in pybamm editing the `__str__` method of `BaseModel` to return `model.name`

File: bot/utils/chemistry_generator.py [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+        ambient_temp: numerical
+        initial_temp: numerical
+        reference_temp: numerical
+    """
+
+    if chemistry == pybamm.parameter_sets.Chen2020:
+        lower_voltage = single_decimal_point(2.5, 4.0, 0.1)
+        ambient_temp = random.uniform(273.18, 298.15)
+        initial_temp = random.uniform(273.18, 298.15)
+        reference_temp = random.uniform(273.18, 298.15)
```
Comment (tinosulzer) [inline bot/utils/chemistry_generator.py:24]: see ref temperature comment above

File: bot/utils/chemistry_generator.py [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+    elif chemistry == pybamm.parameter_sets.Marquis2019:
+        lower_voltage = single_decimal_point(3.1, 3.9, 0.1)
+        ambient_temp = random.uniform(273.18, 298.15)
+        initial_temp = random.uniform(273.18, 298.15)
+        reference_temp = random.uniform(273.18, 298.15)
```
Comment (tinosulzer) [inline bot/utils/chemistry_generator.py:30]: and here

**Review by tinosulzer — APPROVED** [review]

**Review by tinosulzer — APPROVED** [review]
Looks good to me with some minor style changes

**Pull request merged**